intel_extension_for_transformers/llm/runtime/graph/core/layers/mha_dense.cpp

```diff
@@ -303,109 +302,76 @@ class PackedWeightBatch : public jblas::prologue::gemm::WeightBase, public jblas
 template <class GemmCore_T, JBLAS_ISA ISA_T, bool IsTrans, typename T_SRC = typename GemmCore_T::BType>
 class WeightPackBatchBf16Base {
  public:
-  using WType = typename GemmCore_T::BType;  // weight type
-  using SType = T_SRC;                       // source type (before packed)
-  using PW_T = PackedWeightBatch<WType>;     // packed weight type
-  using Parallel = parallel::Parallel2DRowMajor;
+  using WType = typename GemmCore_T::BType;      // weight type
+  using SType = T_SRC;                           // source type (before packed)
+  using StorageType = StoragePackedWeightBatch;  // packed weight type
 
   struct Param {
-    const jblas::prologue::gemm::WeightBase* packedW;
-  };
-
-  // additional parameter to pack weight at runtime
-  struct PackParam {
-    SType* src;
-    int ld;
-    std::function<int(int)> step_batch;
-    int K;
-    int N;
+    const SType* B;
+    const int ldb;
+    const StorageType* packedW;
   };
 
   JBLAS_CODE getWeight(...) = delete;
 
   JBLAS_CODE getWeight(WType** dstptr, int* dststep, int /* b_size */, int /* k_size */, int /* n_size */, int b_offset,
-                       int k_offset, int n_offset, const jblas::prologue::gemm::WeightBase* ptr) {
-    const auto wptr = dynamic_cast<const PW_T*>(ptr);
+                       int k_offset, int n_offset, const Param& param, void* /* tmpcache */, size_t /* cachesize */) {
```

Comment:
b_size not used?

Reply:
Yes. All sizes are not used as this WeightPack class was designed to packed with paddings and should never be runtime copy and padded. Therefore, a pointer to the storage (with some offset) is given directly and the `b/k/n_size` and the `tmpcache` parameters are just here for a consistent API.